src/executables/util_service.c

```diff
@@ -1442,6 +1396,61 @@ process_service (int command_type, bool process_window_service)
   return status;
 }
 
+/*
+ * get_server_names -
+ *
+ * return:
+ *
+ *      out_buffer (out):
+ */
+static void
+get_server_names (char *out_buffer)
+{
+  FILE *input;
+  char buf[4096];
+  char cmd[PATH_MAX];
+
+  if (out_buffer == NULL)
+    {
+      return;
+    }
+
+  make_exec_abspath (cmd, PATH_MAX, (char *) UTIL_COMMDB_NAME " " COMMDB_ALL_STATUS);
+  input = popen (cmd, "r");
+  if (input == NULL)
+    {
+      return;
+    }
+
+  int offset = 0;
+  /* *INDENT-OFF* */
+  std::string delimiter = " ";
+  /* *INDENT-ON* */
+  while (fgets (buf, 4096, input) != NULL)
+    {
+      /* *INDENT-OFF* */
+      std::string s (buf);
+
+      /* ignore HA-Server */
+      if (s.find (CHECK_HA_SERVER) != std::string::npos)
+      {
+	    continue;
+      }
+
+      /* find Server */
+      size_t start = s.find (delimiter, 1) + 1;
+      size_t end = s.find (delimiter, start);
+
+      std::string server_name = s.substr (start, end - start);
+
+      snprintf (out_buffer + offset, 4096, "%s,", server_name.c_str ());
```

Comment:
Since you don't know the out buffer size, out_buffer overflow can occur if you have many ha servers.
You need to add code to prevent overflow.